source/lib/ze_libapi.cpp

```diff
@@ -196,8 +196,8 @@ zeInitDrivers(
                                                     ///< including ::ze_init_driver_type_flag_t combinations.
     )
 {
-    static ze_result_t result = ZE_RESULT_SUCCESS;
-    std::call_once(ze_lib::context->initOnceDrivers, [desc]() {
+    ze_result_t result = ZE_RESULT_SUCCESS;
+    std::call_once(ze_lib::context->initOnceDrivers, [desc,&result]() {
```

Comment:
`call_once` will not call for the second time and `result` will not be updated, `desc` will not be initialized

Reply:
This is meant to be called and set only once by design.

On the first call to zeInitDrivers, the context is init which uses the desc in the call and sets the result.

If this first call fails init, then the if conditional after this "call once" will fail.

This patch removes the static condition that was left over from zeInit() such that the result is allowed to change each time and is not stored since zeInitDrivers is mean to be called more than once.

Reply:
The second call to zeInitDrivers will skip the call_once block because the context does not need to be reInit and the call is meant to proceed to calling the function pointer for zeInitDrivers if it is set by the drivers.

The result needed to be reset in this second call allowing for the second call to succeed where the first failed.

Reply:
> The second call to zeInitDrivers will skip the call_once block because the context does not need to be reInit

who said that the context is the same?

Reply:
There is only one loader lib context. It is guaranteed to be the same and there is only one per loader in a process.

Reply:
I am resolving this issue because there is no problem here with the context. The Patch here successfully resolves the issue where we have a failure on the first call, but the second call would work. 

Please contact me directly if you do not agree.